File: fapi/src/main/java/com/iquiz/fapi/models/Category.java
package com.iquiz.fapi.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Category {

    private int idCategory;
    private String text;
    private String category;
    private String imgLink;

    public Category() {
    }
    public Category(int idCategory, String text, String category, String imgLink) {
        this.idCategory = idCategory;
        this.text = text;
        this.category = category;
        this.imgLink = imgLink;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public int getIdCategory() {
        return idCategory;
    }

    public void setIdCategory(int idCategory) {
        this.idCategory = idCategory;
    }

    public String getImgLink() {
        return imgLink;
    }

    public void setImgLink(String imgLink) {
        this.imgLink = imgLink;
    }
}

File: frontend/src/app/modules/menu/components/category/category.ts
export class Category{
   idCategory: string;
   text: string;
   category: string;
   imgLink: string;

  static cloneBase(category: Category): Category {
    const clonedCategory: Category = new Category();
    clonedCategory.idCategory = category.idCategory;
    clonedCategory.text = category.text;
    clonedCategory.category = category.category;
    clonedCategory.imgLink = category.imgLink;
    return clonedCategory;
  }
}
export class CategoryStr {
}
